internal/singleflight/singleflight/singleflight.go

```diff
@@ -0,0 +1,69 @@
+//
```

Comment:
do you plan to move this package to `internal/singleflight` just before merging?

Reply:
yes.
if author approves this PR and benchmark result, I change from the old singleflight package to the new singleflight package.

Reply:
but how to manage the bench test code? I think you need to change the test case name?

Reply:
Of course the test case name will change. Just replace the implementation. Benchmark code and all results will be erased.

Reply:
ok!

Reply:
please leave `todo` list to the PR description :)